### PR TITLE
Feature/107-108 create contact COM/CAN lookup

### PR DESCRIPTION
**Pull request by dheitzer**
## Summary

- Issue api#<number> and/or app#<number>
  
(Include a summary of proposed changes)

## How to test

(Include any information that may be helpful to the reviewer(s). This might include links to sample pages to test or any local environmental setup that is unusual such as environment variable (never credentials), API version to point to, etc)

-

### CODE REVIEW

**Review by mjtravers — APPROVED** [review]

File: django-backend/fecfiler/contacts/views.py [hunk 1]
```diff
@@ -58,6 +62,72 @@ class ContactViewSet(CommitteeOwnedViewSet):
...
+
+        max_fecfile_results = self.get_int_param_value(
+            request,
+            "max_fecfile_results",
+            default_max_fecfile_results,
+            max_allowed_results,
+        )
+
+        query_params = urlencode({"q": q, "api_key": FEC_API_KEY})
+        url = "{url}?{query_params}".format(
```
Comment (toddlees) [inline django-backend/fecfiler/contacts/views.py:83]: no need to homebrew this.  requests has got you covered https://requests.readthedocs.io/en/latest/user/quickstart/#passing-parameters-in-urls
```
params = urlencode({"q": q, "api_key": FEC_API_KEY}
requests.get(FEC_API_CANDIDATE_LOOKUP_ENDPOINT, params=params).json()
```

File: django-backend/fecfiler/contacts/views.py [hunk 2]
```diff
@@ -58,6 +62,72 @@ class ContactViewSet(CommitteeOwnedViewSet):
...
+    def candidate_lookup(self, request):
+        q = request.GET.get("q")
+        if q is None:
+            return HttpResponseBadRequest()
+
+        max_fec_results = self.get_int_param_value(
+            request, "max_fec_results", default_max_fec_results, max_allowed_results
+        )
+
+        max_fecfile_results = self.get_int_param_value(
```
Comment (toddlees) [inline django-backend/fecfiler/contacts/views.py:75]: this code is repeated a handful of times in this file

Comment (toddlees) [reply django-backend/fecfiler/contacts/views.py:75]: https://github.com/fecgov/fecfile-web-api/pull/357

File: django-backend/fecfiler/contacts/views.py (hunk 1 above)
Comment (toddlees) [reply django-backend/fecfiler/contacts/views.py:83]: https://github.com/fecgov/fecfile-web-api/pull/357

**Review by toddlees — CHANGES_REQUESTED** [review]
suggested changes in [patch](https://github.com/fecgov/fecfile-web-api/pull/357)

**Review by mjtravers — APPROVED** [review]

File: django-backend/fecfiler/contacts/views.py (hunk 1 above)
Comment (mjtravers) [reply django-backend/fecfiler/contacts/views.py]: Resolved with patch

File: django-backend/fecfiler/contacts/views.py (hunk 2 above)
Comment (mjtravers) [reply django-backend/fecfiler/contacts/views.py]: Resolved with patch

**Pull request merged**